internal/worker/sshserver/authentication.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"context"
+	"encoding/base64"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/juju/juju/core/virtualhostname"
+	"github.com/juju/juju/rpc/params"
+)
+
+type authenticatedViaPublicKey struct{}
```

Comment:
in #/19358 you introduced a type for these keys..

Reply:
Yeah, either are fine, but I think the struct approach is possibly more common in the codebase so I'll stick with this and revert the change in the other PR.